app/code/Magento/Catalog/Model/Product/Image.php

```diff
@@ -632,6 +633,26 @@ public function saveFile()
      */
     public function getUrl()
     {
+        if ($this->imageAsset->getMediaFormatUrl() == CatalogMediaConfig::HASH) {
```

Comment:
Hi @SilinMykola. Thank you for your collaboration. I would like to reflect together on the solution you've proposed. If we are talking about GraphQL issue, I would suggest providing the fix on the GraphQL level (resolvers, services connected to resolvers). As a rule, we don't have to change the domain logic to fix GraphQL issues. By changing the domain logic we bring much more potential issues not only for GraphQL but for entire system. 
In this particular case, we may expect a performance degradation for every `getUrl` call (and not only for GraphQL)

Reply:
Hi @rogyar. Thanks for your help! I understood what you are talking about. I should move my solution to resolvers. But I'm afraid that with this approach I will have to add many new dependencies to resolver and call the url formation again. I think maybe it might be worth creating a separate service to check for the existence of a file and build a link.